### ISSUE TITLE
Pings random sites like walmart.com

### ISSUE THREAD

**Issue by madduck** [xbmc/Kore#434]
Hey,

I'm using the latest Kore from F-Droid on a reasonably secured phone, including AFWall+ to control which connections I accept.

I've noticed that Kore pings a number of IPs at random times, e.g.

- 32.57.185.37 (belonging to AT&T)
- 161.164.44.244 (walmart.com)

I can't seem to find a setting for this, nor any documentation, or anything on the Web.

At the very least, this should be configurable, but I'd also be inclined to believe that this is "functionality" which doesn't belong in Kore.

Thanks for your consideration!

**Comment by MartijnKaijser** [comment]
I seriously doubt this is even coming Kore cause wth would we even include that?

**Comment by madduck** [comment]
I don't know, but AFWall+ is logging these parcels as coming from the UID that Kore runs as, @MartijnKaijser. But yeah, I find it hard to believe too!

**Comment by madduck** [comment]
There's another symptom, and it might be related. I'll rename the issue once we know more.

AFWall+ has the option to allow apps to (a) access the LAN, and (b) access Wifi. These two are mutually exclusive. The way this is implemented is through the following two `iptables` lines, `192.168.14.0/24` being my LAN.:

    -A afwall-wifi-fork -d 192.168.14.0/24 -j afwall-wifi-lan
    -A afwall-wifi-fork ! -d 192.168.14.0/24 -j afwall-wifi-wan

I initially only enabled LAN access, as that's all that's needed: phone and Kodi are on the same subnet, as is the DNS server. However, use of the remote is almost impossible, most clicks get lost and if anything, then actions take place multiple seconds after initiating them on Kore.

So I enabled Wifi+LAN access, and now it all works. Obviously, the Ping warnings also go away, but the router still sees ICMP requests from the phone to the above IPs, and only while Kore is open.

**Comment by SyncedSynapse** [comment]
Obviously Kore doesn't send anything to those IPs, unless you specify them as the address of your Kodi server...

You don't have to believe me, just get the code and check for yourself. It's not that much and a couple of grep's should clarify it.

So could you please double check what you're seeing, and if you're still convinced about that, provide some more details, like, in what situations do you see that traffic (everytime Kore is open?), with what interval? what kind of packets is the firewall logging (ICMP echo requests?).

Also, use the binaries available at http://mirrors.kodi.tv/tools/kore/ . I don't have any reason to suspect that F-Droid's version isn't correct, but we aren't responsible for building and supplying it, F-Droid does it directly, and i would prefer if you used a binary that we control its build process (alternatively, build it from source).

I just see 2 ways for this to happen:
1. Your firewall is confusing some valid local traffic and reporting it wrongly.
2. Kore uses some 3rd party open source libraries, and i haven't gone through all the code obviously, but i very much doubt that one of them would be doing that, nevertheless...

**Comment by madduck** [comment]
I've just spent an hour trying to get to the bottom of this. First, I discovered that I could not actually detect the packets at the router, even though they were showing up in the AFWall+ logs when blocking was enabled.

However, I then installed a rule like this:

    iptables -I afwall-wifi-wan -m owner --uid-owner 10188 -j LOG --log-prefix '[Kore] '

and it revealed that in fact, there weren't any packets to walmart.com or any other site being generated.

So my current suspicion now is that AFWall+ conflates things here. I'll report a bug there. Sorry to have taken up your time.
